week4-ts/src/pages/MainPage.tsx

```diff
@@ -0,0 +1,92 @@
+import styled from "styled-components";
+import SearchSection from "../components/SearchSection";
+import ResultSection from "../components/ResultSection";
+import { useState } from "react";
+import { Store } from "../types";
+import { getLocationBasedSearch, getStoreBasedTown } from "../services";
+
+export default function MainPage() {
+  const [storeInfo, setStoreInfo] = useState<Store[]>([]);
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onClickSearchButton = (isCheck: boolean, myTown: string) => {
```

Comment:
브라우저가 새로고침되는 이벤트 디폴트 동작을 막기 위해 
```suggestion
  const onClickSearchButton = (isCheck: boolean, myTown: string, e:React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
```
2. 이런 식으로 event를 받아서 `e.preventDefault();`를 통해 디폴트 이벤트를 막아주면 좋을 것 같습미다!